lib/BC/SleighLifter.cpp

```diff
@@ -0,0 +1,1292 @@
+               llvm::Intrinsic::uadd_with_overflow, overloaded_types);
+           llvm::Value *uadd_args[] = {lhs, rhs};
+           llvm::Value *uadd_val = bldr.CreateCall(uadd_intrinsic, uadd_args);
+           // The value at index 1 is the overflow bit.
+           return bldr.CreateExtractValue(uadd_val, {1});
+         }},
+        {OpCode::CPUI_INT_SCARRY,
+         [](llvm::Value *lhs, llvm::Value *rhs, llvm::IRBuilder<> &bldr) {
+           llvm::Type *overloaded_types[1] = {lhs->getType()};
+           llvm::Function *sadd_intrinsic = llvm::Intrinsic::getDeclaration(
+               bldr.GetInsertBlock()->getModule(),
+               llvm::Intrinsic::sadd_with_overflow, overloaded_types);
+           llvm::Value *sadd_args[] = {lhs, rhs};
+           llvm::Value *sadd_val = bldr.CreateCall(sadd_intrinsic, sadd_args);
+           // The value at index 1 is the overflow bit.
+           return bldr.CreateExtractValue(sadd_val, {1});
+         }},
+        {OpCode::CPUI_INT_SBORROW,
+         [](llvm::Value *lhs, llvm::Value *rhs, llvm::IRBuilder<> &bldr) {
+           llvm::Type *overloaded_types[1] = {lhs->getType()};
+           llvm::Function *ssub_intrinsic = llvm::Intrinsic::getDeclaration(
+               bldr.GetInsertBlock()->getModule(),
+               llvm::Intrinsic::ssub_with_overflow, overloaded_types);
+           llvm::Value *ssub_args[] = {lhs, rhs};
+           llvm::Value *ssub_val = bldr.CreateCall(ssub_intrinsic, ssub_args);
+           // The value at index 1 is the overflow bit.
+           return bldr.CreateExtractValue(ssub_val, {1});
+         }},
+};
+
+SleighLifter::SleighLifter(const sleigh::SleighArch *arch_,
+                           const IntrinsicTable &intrinsics_)
+    : InstructionLifter(arch_, intrinsics_),
+      sleigh_context(new sleigh::SingleInstructionSleighContext(
+          arch_->GetSLAName(), arch_->GetPSpec())) {
+  arch_->InitializeSleighContext(*sleigh_context);
+}
+
+LiftStatus
+SleighLifter::LiftIntoBlock(Instruction &inst, llvm::BasicBlock *block,
+                            llvm::Value *state_ptr, bool is_delayed) {
+
+  if (!inst.IsValid()) {
+    LOG(ERROR) << "Invalid function" << inst.Serialize();
+    inst.operands.clear();
+    return kLiftedInvalidInstruction;
+  }
+
+
+  llvm::IRBuilder<> ir(block);
+  const auto next_pc_ref =
+      LoadRegAddress(block, state_ptr, kNextPCVariableName);
+  const auto next_pc = ir.CreateLoad(this->GetWordType(), next_pc_ref);
+  auto pc_ref = this->LoadRegAddress(block, state_ptr, "PC");
+
+  auto curr_eip = ir.CreateAdd(
+      next_pc, llvm::ConstantInt::get(this->GetWordType(), inst.bytes.size()));
+  ir.CreateStore(curr_eip, next_pc_ref);
+  ir.CreateStore(curr_eip, pc_ref);
+
+
+  SleighLifter::PcodeToLLVMEmitIntoBlock lifter(
+      block, state_ptr, inst, *this, this->sleigh_context->getUserOpNames());
+  auto res = sleigh_context->oneInstruction(inst.pc, lifter, inst.bytes);
+
+  lifter.finalize();
+
+  (void) res;
```

Comment:
`std::ignore = ...` is the way to do it, although it is a red flag unless good reason (there is no comment right now).